syntax = "proto3";

package pb;

option go_package = "./pb";

message RecommendationRequestForUserId {
  string user_id = 1;
  uint64 skip = 2;
  uint64 take = 3;
}

message RecommendationRequestOnViews {
  repeated string ids = 1;
  uint64 skip = 2;
  uint64 take = 3;
}

message ProductReplica {
  string id = 1;
  string name = 2;
  string description = 3;
  double price = 4;
}

message RecommendationResponse {
  repeated ProductReplica recommended_products = 1;
}

service RecommenderService {
  rpc GetRecommendations (RecommendationRequestForUserId) returns (RecommendationResponse);
  rpc GetRecommendationsBasedOnViewed (RecommendationRequestOnViews) returns (RecommendationResponse);
}
